### ISSUE TITLE
Feedback homework week 3

### ISSUE THREAD

**Issue by remarcmij** [Sob7i/hyf-javascript3#3]
Hi Sobhi, here is my feedback on your homework for week 3.

Your homework as submitted does not work. Your JavaScript file is named `app.js` but your `index.html` file tries to load `script.js`, and fails. How can it be that you did not notice this?

If I correct this mistake and load your `index.html` in the browser it loads OK and seems to run OK. However, while it is somewhat responsive, the layout is completely off when I view on a mobile phone screen size (see image).

![sobhi](https://user-images.githubusercontent.com/2788771/38937131-c1513ee6-4322-11e8-9b0e-a908c0973aca.png)

**1.** You didn't implement ES6 classes, for now that is OK. It will be covered again in the React module, which is still some time away.

**2.** You did not sort the list repos displayed in the select box as was asked in Step 4 of the homework.

**3.** On line 168 of app.js you have a line `window.onload = ReposSelect;`. As you should be able to see, this line is green underlined by ESLint. If I hover my mouse over it, I get the message 'Unreachable code.' This means that that code will never be executed. Any code following a `return` statement is unreachable. I assume you intended this code to be outside of the `fetchJSON()` function body.

**4.** You have moved the `fetchJSON()` call, now in line 5, outside of your function `main()`, possibly because the problem that issue 1 above gave you. Furthermore, you didn't use `async`/`await` here. The function call should be moved inside the function `main()`:

```js
async function main() {
  const root = document.getElementById('root');
  //...
  try {
    const repos = await fetchJSON(urlAPI);
    repos.forEach(repo => {
      createAndAppend('option', select, { html: repo.name });
    });
  } catch (error) {
    root.innerHTML = 'Error:' + ' ' + error.message;
  }
}
```

Now, at the end of your file, you can add a statement:

```js
window.onload = main;
```

**5.** You need to be more clear in how you name variables and parameters. For instance in line 39:

```js
data.forEach(data => {
  //...
});
```

What do you mean with `data`? You might as well use the name `stuff`, that would be just as clear (not!). Express yourself clearly, for instance:

```js
repos.forEach(repo => {
  //...
});
```

**6.** Function names should be in camelCase: `ReposInfo` -> `reposInfo`. They should also indicate 'action', therefore need a verb in the name, e.g. `renderRepos`. Check the [naming conventions](https://github.com/HackYourFuture/fundamentals/blob/master/fundamentals/naming_conventions.md) fundamental.

Bottom line: I think you should spend more time and effort to ensure that you deliver quality code. Don't start homework on a Saturday afternoon because that will not give enough time to produce something decent, nor time to ask questions. That I could not run your code because of the wrong file name is sloppy to say the least.
